File: app/src/main/java/com/tuvakov/zetube/android/data/SubscriptionDao.java
package com.tuvakov.zetube.android.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface SubscriptionDao {

    @Query("SELECT * FROM subscriptions")
    LiveData<List<Subscription>> selectAll();

    @Insert
    void insert(Subscription subscription);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void bulkInsert(List<Subscription> subscriptions);

    @Update
    void update(Subscription subscription);

    @Delete
    void delete(Subscription subscription);

    @Query("DELETE FROM subscriptions")
    void deleteAll();

}

File: app/src/main/java/com/tuvakov/zetube/android/data/SubscriptionDao.kt
package com.tuvakov.zetube.android.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface SubscriptionDao {

    @get: Query("SELECT * FROM subscriptions ORDER BY title ASC")
    val subscriptions: LiveData<List<Subscription>>

    @Query("SELECT * FROM subscriptions WHERE id = :channelId")
    suspend fun getSubscriptionById(channelId: String): Subscription

    @Insert
    suspend fun insert(subscription: Subscription)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun bulkInsert(subscriptions: List<Subscription>)

    @Update
    suspend fun update(subscription: Subscription)

    @Delete
    suspend fun delete(subscription: Subscription)

    @Query("DELETE FROM subscriptions")
    suspend fun deleteAll()
}